pkg/types/aws/platform.go

```diff
@@ -19,4 +19,20 @@ type Platform struct {
 	// platform configuration.
 	// +optional
 	DefaultMachinePlatform *MachinePool `json:"defaultMachinePlatform,omitempty"`
+
+	// CustomRegionOverride list contains custom endpoints which will overridde default
+	// service endpoint of AWS Services. 
+	// +optional
+	CustomRegionOverride []CustomEndpoint  `json:"customRegionOverride,omitempty"` 
```

Comment:
I'm not sure how I feel about the name `CustomRegionOverride`. I think we borrowed that name from the k8s cloudproviderconfig, and that's fine. I think it is especially clunky in the install config. personally I like just endpoint but also perhaps customEndpoint.